cpp/tests/CMakeLists.txt

```diff
@@ -38,28 +47,97 @@ add_library(cugraphtestutil STATIC
             ../../thirdparty/mmio/mmio.c)
 
 target_compile_options(cugraphtestutil
-            PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUGRAPH_CXX_FLAGS}>"
-                   "$<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:CUDA>:${CUGRAPH_CUDA_FLAGS}>>"
+    PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUGRAPH_CXX_FLAGS}>"
+           "$<$<COMPILE_LANGUAGE:CUDA>:${CUGRAPH_CUDA_FLAGS}>"
 )
 
 set_property(TARGET cugraphtestutil PROPERTY POSITION_INDEPENDENT_CODE ON)
 
 target_include_directories(cugraphtestutil
     PUBLIC
-        "${CMAKE_CURRENT_SOURCE_DIR}/../../thirdparty/mmio"
         "${CMAKE_CURRENT_SOURCE_DIR}"
+        "${CMAKE_CURRENT_SOURCE_DIR}/../../thirdparty/mmio"
+    PRIVATE
         "${CUGRAPH_SOURCE_DIR}/src"
 )
 
 target_link_libraries(cugraphtestutil
     PUBLIC
         cugraph::cugraph
-        NCCL::NCCL
     PRIVATE
-        cuco::cuco
         GTest::gtest
 )
 
+
+###################################################################################################
+# - common C API test utils -----------------------------------------------------------------------
+
+add_library(cugraph_c_testutil STATIC
+            c_api/test_utils.cpp)
+
+target_compile_options(cugraph_c_testutil
+    PUBLIC "$<$<COMPILE_LANGUAGE:C>:${CUGRAPH_C_FLAGS}>"
+           "$<$<COMPILE_LANGUAGE:CXX>:${CUGRAPH_CXX_FLAGS}>"
+           "$<$<COMPILE_LANGUAGE:CUDA>:${CUGRAPH_CUDA_FLAGS}>"
+)
+
+target_include_directories(cugraph_c_testutil
+    PRIVATE
+        "${CUGRAPH_SOURCE_DIR}/src"
+)
+
+set_property(TARGET cugraph_c_testutil PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+target_link_libraries(cugraph_c_testutil
+    PUBLIC
+        CUDA::toolkit
+    PRIVATE
+        cugraph::cugraph_c
+        cugraph::cugraph
```

Comment:
Should this be specified explicitly or should this testutil just depend on cugraph_c and pull cugraph transitively? Does it have a direct dependency on cugraph?

Reply:
It has a direct dependency on cugraph::cugraph

Reply:
cugraph_c doesn't bring in cugraph as a transitive dep as well

Reply:
> Should we have cugraph_c bring in cugraph as a dependency? This was an area of indecision for me when we first added it.

When talking about CMake dependencies ( via `PUBLIC` vs `PRIVATE` ) keyword we need to think about only what is required for correct compilation and linking, and not what is required at runtime. `cugraph_c` is designed as a pure `C` language firewall. No cugraph C++ types are allowed to pass this firewall, and therefore no build/link time consumers of `cugraph_c` should need to know about `cugraph`.

Therefore to ensure this firewall is correct, we want a `PRIVATE` linkage from `cugraph_c` to `cugraph`